tests/kafkatest/services/kafka_log4j_appender.py

```diff
@@ -67,13 +67,9 @@ def start_cmd(self, node):
 
     def stop_node(self, node):
         node.account.kill_process("VerifiableLog4jAppender", allow_fail=False)
-        if self.worker_threads is None:
-            return
 
-        # block until the corresponding thread exits
-        if len(self.worker_threads) >= self.idx(node):
-            # Need to guard this because stop is preemptively called before the worker threads are added and started
-            self.worker_threads[self.idx(node) - 1].join()
+        stopped = self.wait_node(node, timeout_sec=self.stop_timeout_sec)
+        assert stopped, "Node did not stop within the specified timeout of %s seconds" % str(self.stop_timeout_sec)
```

Comment:
Is there any information from the node itself that we could add to this message so that we have an idea which instance failed to shutdown properly?